.github/workflows/deploy.yml

```diff
@@ -0,0 +1,44 @@
+name: Deploy Website
+
+on:
+  push:
+    branches: [ 'master' ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: '14.x'
+      - name: Install dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build:ci
+      - name: Archive build
+        if: success()
+        uses: actions/upload-artifact@v1
+        with:
+          name: deploy_dist
+          path: dist
+  deploy:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+      - name: Download build
+        uses: actions/download-artifact@v1
+        with:
+          name: deploy_dist
+      - name: Deploy to GitHub Pages
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Where are the secrets? 👀

Reply:
Not 100% certain, but afaik it's the GitHub Access Token for whoever pushed the commit to master.

This action takes the previously built files and pushes them to the `gh-pages` branch. You can see this below:
https://github.com/UCSolarCarTeam/UCSolarCarTeam.github.io/tree/gh-pages

Notice how the latest commit shows that I was the author. I technically was but I never created the commit, as it was automatically created by this action.